### PR TITLE
CLOUDP-114669: Automatically add trailing slash to the URL

### PR DESCRIPTION
**Pull request by igor-karpukhin**
## Description

This PR adds an ability for the `Client` to automatically add trailing slash (`/`) to the end of the provided URL. The motivation behind this is that some internal tools use this library and it is easier to assume that the library handles both cases: with, and without trailing slash.

cc: @jamesbroadhead 

Link to any related issue(s): https://jira.mongodb.org/browse/CLOUDP-114669

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

**Review by andreaangiolillo — APPROVED** [review]
LGTM

**Comment by igor-karpukhin** [comment]
@andreaangiolillo thanks for approving this PR. Could you please approve CI/CD workflow if you have one?

**Review by gssbzn — DISMISSED** [review]
Before merging this we nened this chnages on the ops manager client

**Comment by gssbzn** [comment]
We keep atlas and ops manager compatible so we need this same change for https://github.com/mongodb/go-client-mongodb-ops-manager

**Comment by andreaangiolillo** [comment]
FYI: https://github.com/mongodb/go-client-mongodb-ops-manager/pull/142

**Pull request merged**
